ert/data/record/_record.py

```diff
@@ -59,19 +59,51 @@ class RecordType(str, Enum):
 
 
 class Record(ABC):
+    """The :py:class:`Record` class is an abstract class that
+    includes record data and the record type.
+
+    The record needs be either `BlobRecord` or `NumericalRecord` type derived
+    from `Record` class
+    """
+
     @property
     @abstractmethod
     def data(self) -> record_data:
+        """Returns record data.
+
+        Returns:
+            record_data: record data as
+            Union[numerical_record_data, blob_record_data]
+        """
         pass
 
     @property
     @abstractmethod
     def record_type(self) -> RecordType:
+        """Returns the type of the record data.
```

Comment:
It is possible to encode the available return types as some kind of enum?

Reply:
I have removed docstrings for abstract methods.